src/commands.ts

```diff
@@ -317,7 +309,36 @@ export class CosmosDBCommands {
         }
         return null;
     }
-
+    public static async dropDocDBDatabase(db: DocDBDatabaseNode, explorer: CosmosDBExplorer): Promise<void> {
+        const confirmed = await vscode.window.showWarningMessage("Are you sure to delete database: " + db.label + " and its collections?",
+            "Yes", "No");
+        if (confirmed === "Yes") {
+            const masterKey = await db.getPrimaryMasterKey();
+            const endpoint = await db.getEndpoint();
+            const client = new DocumentClient(endpoint, { masterKey: masterKey });
+            await new Promise((resolve, reject) => {
+                client.deleteDatabase(db.getDbLink(), function (err, result) {
+                    err ? reject(err.body) : resolve(result);
```

Comment:
This 'reject' won't ever be displayed to the user. You could do this:
```
reject(new Error(err.body))
```
But I would actually modify extension.ts so that you don't have to fix other cases:
```
else if (typeof err === "string") {
    vscode.window.showErrorMessage(err);
}
```

(Or you could do both)